wkcuber/api/Layer.py

```diff
@@ -88,6 +105,74 @@ def set_bounding_box_size(self, size: Tuple[int, int, int]):
         for _, mag in self.mags.items():
             mag.view.size = size
 
+    def setup_mag(self, mag: Mag) -> None:
+        raise NotImplemented
+
+    def _initialize_mag_from_other_mag(self, new_mag_name, other_mag, compress):
+        raise NotImplemented
+
+    def downsample(
+            self,
+            from_mag: Mag,
+            max_mag: Mag,
+            interpolation_mode: str,
+            compress: bool,
+            scale: Tuple[float, float, float] = None,
+            buffer_edge_len=None,
+            args: Namespace = None,
+    ):
+        # if 'scale' is set, the data gets downsampled anisotropic
+        interpolation_mode = parse_interpolation_mode(interpolation_mode, self.name)
```

Comment:
I copied this from the old implementation, but I would assume that the second parameter should be the type of the layer instead of the name, right?

Reply:
Yes, you are completely right. Good catch! Seems like mode-filtering was used for all layers which weren't named `color` in the past :see_no_evil: